server/controllers/finance/purchases.js

```diff
@@ -361,58 +362,82 @@ function find(options) {
  * if it is completed or partially entered
  */
 function purchaseStatus(req, res, next) {
-  const status = {};
   const FROM_PURCHASE_ID = 1;
+  const PURCHASE_PARTIALLY_RECEIVED = 4;
+  const PURCHASE_FULLY_RECEIVED = 3;
+  const PURCHASE_EXCESSIVELY_RECEIVED = 6;
+
+  const glb = {};
   const purchaseUuid = db.bid(req.params.uuid);
-  const sql = `
-    SELECT IFNULL(SUM(m.quantity * m.unit_cost), 0) AS movement_cost, p.cost, CEIL(DATEDIFF(m.date, p.date)) AS delay
+
+  const sqlQuantities = `
+    SELECT SUM(t.purchase_quantity) AS purchase_quantity, SUM(t.movement_quantity) AS movement_quantity
+    FROM (
+      (
+        SELECT IFNULL(SUM(pi.quantity), 0) AS purchase_quantity, 0 AS movement_quantity FROM purchase p
+        JOIN purchase_item pi ON pi.purchase_uuid = p.uuid
+        WHERE p.uuid = ?
+      ) UNION ALL (
+        SELECT 0 AS purchase_quantity, SUM(m.quantity) AS movement_quantity FROM stock_movement m 
+        JOIN lot l ON l.uuid = m.lot_uuid 
+        JOIN purchase p ON p.uuid = l.origin_uuid
+        WHERE p.uuid = ? AND m.flux_id = ? AND m.is_exit = 0
+      )
+    )t
+  `;
+
+  const sqlPurchaseDelay = `
+    SELECT
+      p.cost, CEIL(DATEDIFF(m.date, p.date)) AS delay
     FROM stock_movement m
     JOIN lot l ON l.uuid = m.lot_uuid
     JOIN purchase p ON p.uuid = l.origin_uuid
-    WHERE p.uuid = ? AND m.flux_id = ? AND m.is_exit = 0;
+    WHERE p.uuid = ? AND m.flux_id = ? AND m.is_exit = 0 LIMIT 1;
```

Comment:
If you are only taking one of these, should you have a ORDER BY so that we know which one you are taking...?